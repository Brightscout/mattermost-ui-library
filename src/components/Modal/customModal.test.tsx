import React from 'react';
import {mount, ReactWrapper} from 'enzyme';

import CustomModal from '.';

// Mock props data
const customModalChildrenClass = 'custom-modal-children';
const customModalChildren = <div className={customModalChildrenClass}>{'Custom Modal'}</div>;
const customModalProps = {
    show: true,
    onHide: jest.fn,
    className: 'custom-modal-class',
};

describe('CustomModal', () => {
    let component: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = mount(<CustomModal {...customModalProps}>{customModalChildren}</CustomModal>);
    });

    afterEach(() => {
        component.unmount();
    });

    it('Should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should apply the passed className prop', () => {
        expect(component.hasClass(customModalProps.className)).toBeTruthy();
    });

    it('Should render the children correctly', () => {
        const modalChildren = document.querySelector(`.${customModalChildrenClass}`);
        expect(modalChildren).toBeInTheDocument();
    });
});
