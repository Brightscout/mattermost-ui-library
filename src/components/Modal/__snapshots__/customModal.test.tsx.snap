// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomModal Should render correctly 1`] = `
<Modal
  className="custom-modal-class"
  onHide={[Function]}
  show={true}
>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    centered={true}
    className="modal custom-modal-class"
    dialogAs={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "ModalDialog",
        "render": [Function],
      }
    }
    enforceFocus={true}
    keyboard={true}
    onHide={[Function]}
    restoreFocus={true}
    show={true}
  >
    <Modal
      autoFocus={true}
      backdrop={true}
      backdropTransition={[Function]}
      enforceFocus={true}
      keyboard={true}
      manager={
        BootstrapModalManager {
          "handleContainerOverflow": true,
          "isRTL": false,
          "modals": Array [
            Object {
              "add": [Function],
              "backdrop": <div
                class="fade modal-backdrop show"
              />,
              "dialog": <div
                aria-modal="true"
                class="fade modal custom-modal-class modal show"
                role="dialog"
                style="display: block; padding-right: 0px;"
                tabindex="-1"
              >
                <div
                  class="modal-dialog modal-dialog-centered"
                >
                  <div
                    class="modal-content"
                  >
                    <div
                      class="modal__header  modal-header"
                    >
                      <div
                        class="modal__title d-flex align-items-center justify-content-between"
                      >
                        <button
                          class="style--none header-close-btn plugin-btn"
                        >
                          <i
                            class="icon icon-close modal__close-icon cursor-pointer font-20"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="modal__loader-container position-relative "
                    />
                    <div
                      class="modal__body padding-v-15 padding-h-12  modal-body"
                    >
                      <div
                        class="custom-modal-children"
                      >
                        Custom Modal
                      </div>
                    </div>
                    <div
                      class="modal__footer d-flex flex-row-reverse align-items-center  padding-h-12 padding-v-15 modal-footer"
                    >
                      <button
                        class="btn btn-link modal__cancel-btn margin-0 padding-0 plugin-btn"
                      >
                        Cancel
                      </button>
                    </div>
                  </div>
                </div>
              </div>,
              "isTopModal": [Function],
              "remove": [Function],
              "setBackdropRef": [Function],
              "setDialogRef": [Function],
            },
          ],
          "ownerDocument": undefined,
          "state": Object {
            "scrollBarWidth": 1024,
            "style": Object {
              "overflow": "",
              "paddingRight": "",
            },
          },
        }
      }
      onEnter={[Function]}
      onEntering={[Function]}
      onEscapeKeyDown={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onHide={[Function]}
      renderBackdrop={[Function]}
      renderDialog={[Function]}
      restoreFocus={true}
      show={true}
      transition={[Function]}
    >
      <Portal
        containerInfo={
          <body
            class="modal-open"
            data-rr-ui-modal-open=""
            style="padding-right: 1024px;"
          >
            <div
              class="fade modal-backdrop show"
            />
            <div
              aria-modal="true"
              class="fade modal custom-modal-class modal show"
              role="dialog"
              style="display: block; padding-right: 0px;"
              tabindex="-1"
            >
              <div
                class="modal-dialog modal-dialog-centered"
              >
                <div
                  class="modal-content"
                >
                  <div
                    class="modal__header  modal-header"
                  >
                    <div
                      class="modal__title d-flex align-items-center justify-content-between"
                    >
                      <button
                        class="style--none header-close-btn plugin-btn"
                      >
                        <i
                          class="icon icon-close modal__close-icon cursor-pointer font-20"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="modal__loader-container position-relative "
                  />
                  <div
                    class="modal__body padding-v-15 padding-h-12  modal-body"
                  >
                    <div
                      class="custom-modal-children"
                    >
                      Custom Modal
                    </div>
                  </div>
                  <div
                    class="modal__footer d-flex flex-row-reverse align-items-center  padding-h-12 padding-v-15 modal-footer"
                  >
                    <button
                      class="btn btn-link modal__cancel-btn margin-0 padding-0 plugin-btn"
                    >
                      Cancel
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </body>
        }
      >
        <BackdropTransition
          appear={true}
          in={true}
        >
          <Fade
            appear={true}
            in={true}
            mountOnEnter={false}
            timeout={null}
            unmountOnExit={false}
          >
            <ForwardRef
              addEndListener={[Function]}
              appear={true}
              childRef={[Function]}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              timeout={null}
              unmountOnExit={false}
            >
              <Transition
                addEndListener={[Function]}
                appear={true}
                enter={true}
                exit={true}
                in={true}
                mountOnEnter={false}
                nodeRef={
                  Object {
                    "current": <div
                      class="fade modal-backdrop show"
                    />,
                  }
                }
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={null}
                unmountOnExit={false}
              >
                <div
                  className="fade modal-backdrop show"
                  onClick={[Function]}
                />
              </Transition>
            </ForwardRef>
          </Fade>
        </BackdropTransition>
        <DialogTransition
          appear={true}
          in={true}
          onEnter={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          unmountOnExit={true}
        >
          <Fade
            appear={true}
            in={true}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            timeout={null}
            unmountOnExit={true}
          >
            <ForwardRef
              addEndListener={[Function]}
              appear={true}
              childRef={[Function]}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              timeout={null}
              unmountOnExit={true}
            >
              <Transition
                addEndListener={[Function]}
                appear={true}
                enter={true}
                exit={true}
                in={true}
                mountOnEnter={false}
                nodeRef={
                  Object {
                    "current": <div
                      aria-modal="true"
                      class="fade modal custom-modal-class modal show"
                      role="dialog"
                      style="display: block; padding-right: 0px;"
                      tabindex="-1"
                    >
                      <div
                        class="modal-dialog modal-dialog-centered"
                      >
                        <div
                          class="modal-content"
                        >
                          <div
                            class="modal__header  modal-header"
                          >
                            <div
                              class="modal__title d-flex align-items-center justify-content-between"
                            >
                              <button
                                class="style--none header-close-btn plugin-btn"
                              >
                                <i
                                  class="icon icon-close modal__close-icon cursor-pointer font-20"
                                />
                              </button>
                            </div>
                          </div>
                          <div
                            class="modal__loader-container position-relative "
                          />
                          <div
                            class="modal__body padding-v-15 padding-h-12  modal-body"
                          >
                            <div
                              class="custom-modal-children"
                            >
                              Custom Modal
                            </div>
                          </div>
                          <div
                            class="modal__footer d-flex flex-row-reverse align-items-center  padding-h-12 padding-v-15 modal-footer"
                          >
                            <button
                              class="btn btn-link modal__cancel-btn margin-0 padding-0 plugin-btn"
                            >
                              Cancel
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>,
                  }
                }
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={null}
                unmountOnExit={true}
              >
                <div
                  aria-modal={true}
                  className="fade modal custom-modal-class modal show"
                  onClick={[Function]}
                  onMouseUp={[Function]}
                  role="dialog"
                  style={
                    Object {
                      "display": "block",
                      "paddingLeft": undefined,
                      "paddingRight": 0,
                    }
                  }
                  tabIndex={-1}
                >
                  <ModalDialog
                    centered={true}
                    onMouseDown={[Function]}
                  >
                    <div
                      className="modal-dialog modal-dialog-centered"
                      onMouseDown={[Function]}
                    >
                      <div
                        className="modal-content"
                      >
                        <ModalHeader
                          onHide={[Function]}
                          showCloseIconInHeader={true}
                        >
                          <ModalHeader
                            className="modal__header "
                            closeButton={false}
                            closeLabel="Close"
                          >
                            <ForwardRef
                              className="modal__header  modal-header"
                              closeButton={false}
                              closeLabel="Close"
                            >
                              <div
                                className="modal__header  modal-header"
                              >
                                <div
                                  className="modal__title d-flex align-items-center justify-content-between"
                                >
                                  <button
                                    className="style--none header-close-btn plugin-btn"
                                    onClick={[Function]}
                                  >
                                    <i
                                      className="icon icon-close modal__close-icon cursor-pointer font-20"
                                    />
                                  </button>
                                </div>
                              </div>
                            </ForwardRef>
                          </ModalHeader>
                        </ModalHeader>
                        <ModalLoader
                          loading={false}
                        >
                          <div
                            className="modal__loader-container position-relative "
                          />
                        </ModalLoader>
                        <ModalBody>
                          <ModalBody
                            className="modal__body padding-v-15 padding-h-12 "
                          >
                            <div
                              className="modal__body padding-v-15 padding-h-12  modal-body"
                            >
                              <ModalSubtitleAndError />
                              <div
                                className="custom-modal-children"
                              >
                                Custom Modal
                              </div>
                              <ModalSubtitleAndError />
                            </div>
                          </ModalBody>
                        </ModalBody>
                        <ModalFooter
                          onHide={[Function]}
                        >
                          <ModalFooter
                            className="modal__footer d-flex flex-row-reverse align-items-center  padding-h-12 padding-v-15"
                          >
                            <div
                              className="modal__footer d-flex flex-row-reverse align-items-center  padding-h-12 padding-v-15 modal-footer"
                            >
                              <button
                                className="btn btn-link modal__cancel-btn margin-0 padding-0 plugin-btn"
                                onClick={[Function]}
                              >
                                Cancel
                              </button>
                            </div>
                          </ModalFooter>
                        </ModalFooter>
                      </div>
                    </div>
                  </ModalDialog>
                </div>
              </Transition>
            </ForwardRef>
          </Fade>
        </DialogTransition>
      </Portal>
    </Modal>
  </Modal>
</Modal>
`;
